packages/block-editor/src/store/selectors.js

```diff
@@ -2052,6 +2042,20 @@ export const getInserterItems = createSelector(
 					.map( buildReusableBlockInserterItem )
 			: [];
 
+		if ( syncStatus === 'unsynced' ) {
+			return reusableBlockInserterItems;
+		}
```

Comment:
In hindsight, there should have been some previous feedback against the addition of the `sync_status` argument to this selector.

With this change, it feels like a good idea to go ahead and remove it. The `unsynced` patterns could be returned by a new private selector.